sources/controllers/AssistantController.ts

```diff
@@ -1,10 +1,140 @@
+
+    private readonly timeSlotView: TimeSlotView;
+
+    constructor(
+        // Repositories
+        calendarRepository: CalendarRepository,
+        eventRepository: EventRepository,
+        // Services
+        assistantSlotService: AssistantSlotService,
+        dateService: DateService,
+        // Views
+        timeSlotView: TimeSlotView,
+    ) {
+        this.calendarRepository = calendarRepository;
+        this.eventRepository = eventRepository;
+        this.assistantSlotService = assistantSlotService;
+        this.dateService = dateService;
+        this.timeSlotView = timeSlotView;
+    }
+
+    public readonly findBestSlots = async (req: Request, res: Response) => {
+        const requestingUser = getRequestingUser(req);
+
+        // Validate request's parameters
+        const calendarId = Number.parseInt(req.params.calendarId, 10);
+        if (isNaN(calendarId)) {
+            throw Boom.badRequest(`Invalid Calendar id "${req.params.calendarId}"`);
+        }
+        const slotDuration = Number.parseInt(req.query.duration, 10);
+        if (!slotDuration) {
+            throw Boom.badRequest(`Invalid slot duration "${req.query.duration}"`);
+        }
+        const slotMinDate = new Date(req.query.min_time);
+        if (!this.dateService.isValidDate(slotMinDate)) {
+            throw Boom.badRequest(`Invalid min_time "${req.query.min_time}"`);
+        }
+        const slotMaxDate = new Date(req.query.max_time);
+        if (!this.dateService.isValidDate(slotMaxDate)) {
+            throw Boom.badRequest(`Invalid max_time "${req.query.max_time}"`);
+        }
+        const eventType: string | undefined = req.query.type;
+        if (!eventType) {
+            throw Boom.badRequest('Missing "type"');
+        }
+        const nbSlots = Number.parseInt(req.query.limit || AssistantController.DEFAULT_SLOT_LIMIT, 10);
+        if (isNaN(nbSlots) || nbSlots <= 0) {
+            throw Boom.badRequest(`Invalid limit "${req.query.limit}"`);
+        }
+
+        // Get calendar
+        const calendar = await this.calendarRepository.getCalendarWithMembers(calendarId);
+        if (!calendar) {
+            throw Boom.notFound("Calendar does not exist");
+        }
+        if (!calendar.members) {
+            throw new Error("Calendar should include members");
+        }
+
+        // Make sure requestingUser is a member of the Calendar
+        if (calendar.members.findIndex(m => m.userId === requestingUser.id) < 0) {
+            throw Boom.forbidden("You do not have access to this Calendar");
+        }
+
+        // Get all Event's related to the Calendar members in date range
+        const eventsPromises = Promise.all(calendar.members.map(m => this.eventRepository.getAllEventsForUser(
+            m.userId,
+            {
+                afterDate: slotMinDate,
+                beforeDate: slotMaxDate,
+                onlyStatuses: [
+                    EventAttendeeStatus.NotGoing,
```

Comment:
Why do you fetch the ones where the user is not going ?

Reply:
Hmm, indeed. I wanted the exact opposite